modules/web-mercator/src/web-mercator-utils.js

```diff
@@ -195,25 +195,28 @@ export function getProjectionParameters({
   width,
   height,
   altitude = DEFAULT_ALTITUDE,
+  fovy,
   pitch = 0,
   nearZMultiplier = 1,
   farZMultiplier = 1
 }) {
+  const halfFov = fovy ? 0.5 * fovy : Math.atan(0.5 / altitude);
+  const focalDistance = fovy ? 0.5 / Math.tan(0.5 * fovy * DEGREES_TO_RADIANS) : altitude;
```

Comment:
`focalDistance` should always be `altitude` (aka ground)

Reply:
Are you sure about this? I've only ever come across the focal distance being obtained from the field of view.

Reply:
The projectionMatrix itself does not use this value. focalDistance is used in deck to calculate the scale of features at the target - a viewMatrix parameter - defined using altitude.

Reply:
I think the real question here is whether we want `fovy` and `altitude` to be independent inputs. Looking across the two repos:

- `@math.gl/web-mercator`'s `WebMercatorViewport`:
  According to your updated doc, if a `fovy` parameter is supplied, then `altitude` is ignored. Then `altitude` must be calculated from `fovy` and passed to `getViewMatrix`, instead of using the default value.
- `deck.gl`'s `WebMercatorViewport`:
  If `projectionMatrix` is supplied, then `fovy` is calculated from `projectionMatrix`, while `altitude` is still required.

Reply:
I think part of the friction here is coming from the past decision to model the matrices such that they be compatible with Mapbox. Their choice of matrices mean that `fovy` and `altitude` can be derived from a single parameter. However the same is not true in general and in particular is incompatible with how Google Maps has their matrices set up. In other words, Mapbox has "ground distance" equal to "focal distance", while Google Maps doesn't.

Thus, it is necessary to allow the two parameters to be specified separately. In general, when constructing a perspective projection matrix, we are simulating a camera and `fovy` is a natural parameter which has a physical analogue in a real-world lens. I would argue that `altitude` should not even be used to construct this matrix as the projection doesn't care about camera position. In fact the first thing `getProjectionParameters` does with the `altitude` is convert it into a field of view, but this is a Mapbox specific implementation detail, and not an OpenGL convention.

In the math.gl documentation I made clear that `fovy` only overrides `altitude` in the construction of the projection matrix.

Regarding the `focalDistance`, if it should always be ground (so equal to `altitude`) can we not change deck to use `altitude`? I can update the code in math.gl to always set `focalDistance` to `altitude`, but it feels wrong because it is specific to deck.gl and goes against the widely held relationship of `2 * focalDistance = 1 / tan(fov / 2)`. Perhaps the `focalDistance` variable in deck.gl should in fact be called `groundDistance` and obtained from the view matrix (or equivalently `altitude`)

Reply:
Let me know if we need a 3-way sync on this to make sure you can proceed, or if you are getting to a conclusion.

As general thoughts I agree that this class is much too mapbox specific, and to give some history `focalDistance` is indeed a bit of a hack. Mapbox creates their matrices so that this value is automatically aligned with `altitude`. 

I recall making an attempt to decompose their matrix creation in the past to make it more general, composable and easy to understand but was not able to get the scale factors right in the time I could spare, so I threw in the `focalDistance` parameter.

Also, the focalDistance parameter is causing pain for some users. It mostly affects PointCloudLayer and tends to cause points to be too small to show making users conclude the layer is broken. Another reason to improve this.